src/components/Modal/VotingPowerDelegationDetail/VotingPowerDelegationDetail.tsx

```diff
@@ -60,8 +64,31 @@ function VotingPowerDelegationDetail({ userVotes, candidate, onBackClick }: Voti
   const [isExpanded, setIsExpanded] = useState(false)
   const [matchingVotes, setMatchingVotes] = useState<MatchResult | null>(null)
   const [showFadeout, setShowFadeout] = useState(true)
+  const [isDelegating, setIsDelegating] = useState(false)
   const [filteredCandidateVotes, setFilteredCandidateVotes] = useState<VotedProposal[]>([])
 
+  const [delegatedAddress] = useAsyncMemo(
```

Comment:
I think this hook can be inside useSnapshotDelegateContract and useSnapshotDelegateContract can return the delegatedAddress.